code/dqcommon/savedata.q

```diff
@@ -1,16 +1,23 @@
 \d .dqe
-savedata:{[dir;pt;ns;tabname]
+savedata:{[dir;pt;savetemp;ns;tabname]
   .lg.o[`dqe;"Saving ",(string tabname)," data to ",.os.pth dir];
   pth:` sv .Q.par[dir;pt;tabname],`;
   err:{[e].lg.e[`savedata;"Failed to save dqe data to disk : ",e];'e};
-  .[upsert;(pth;.Q.en[dir;r:0!.save.manipulate[tabname;ns tabname]]);err];
-  .lg.o[`delete;"deleting ",(string tabname)," data from in-memory table"];                                     /- empty the table from memory
+  tab:.Q.dd[ns;tabname];
+  .[upsert;(pth;.Q.en[dir;r:0!.save.manipulate[tabname;{[tab;x]exec from tab where i=x}[tab]'[savetemp] ]]);err];
```

Comment:
is this pulling one row at a time from table? why not something like

```suggestion
  .[upsert;(pth;.Q.en[dir;r:0!.save.manipulate[tabname;select from tab where i in savetemp]]);err];
```

?

Reply:
testing rn - if there is no difference will make the changes